iredis/entry.py

```diff
@@ -233,6 +238,34 @@ def gather_args(ctx, h, p, n, password, newbie, iredisrc, decode, raw, rainbow,
     if rainbow is not None:
         config.rainbow = rainbow
 
+    dsn_uri = None
+
+    if config_obj["alias_dsn"] and dsn:
+        try:
+            dsn_uri = config_obj["alias_dsn"][dsn]
+        except KeyError:
+            click.secho(
+                "Could not find the specified DSN in the config file. "
+                'Please check the "[alias_dsn]" section in your '
+                "iredisrc.",
+                err=True,
+                fg="red",
+            )
+            exit(1)
+
+    if dsn_uri:
+        uri = urlparse(dsn_uri)
```

Comment:
Can we support `rediss://` and unix socket as well?

There is an implementation in redis-py, which is a very good reference.

https://github.com/andymccurdy/redis-py/blob/b2989759d9c4c73d28251b7976d0823770323e0c/redis/connection.py#L941